office/libreoffice/Portfile

```diff
@@ -0,0 +1,328 @@
+use_xz              yes
+
+set full_version    ${version}.1
+
+set addons_uri      https://dev-www.libreoffice.org/src/
+set extern_uri      https://dev-www.libreoffice.org/extern/
+set osuosl_uri      https://ftp.osuosl.org/pub/tdf/${name}/src/${version}/
+
+set main_distfile   ${name}-${full_version}${extract.suffix}
+
+livecheck.type      regex
+livecheck.url       https://downloadarchive.documentfoundation.org/libreoffice/old/latest/src/
+livecheck.regex     "${name}-(\\d+\\.\\d+\\.\\d+)(?:\\.\\d+)?${extract.suffix}</a></td>"
+
+extract.only        ${main_distfile} \
+                    ${name}-dictionaries-7.0.3.1.tar.xz \
+                    ${name}-translations-7.0.3.1.tar.xz
+master_sites        https://download.documentfoundation.org/${name}/src/${version}/:main \
+                    ${extern_uri}:opens \
+                    ${addons_uri}:ucpp \
+                    ${addons_uri}:clucene-core \
+                    ${addons_uri}:mythes \
+                    ${addons_uri}:dtoa \
+                    ${addons_uri}:abw \
+                    ${addons_uri}:cmis \
+                    ${addons_uri}:ebook \
+                    ${addons_uri}:epubgen \
+                    ${addons_uri}:exttextcat \
+                    ${addons_uri}:mwaw \
+                    ${addons_uri}:orcus \
+                    ${addons_uri}:qxp \
+                    ${addons_uri}:libstaroffice \
+                    ${addons_uri}:zmf \
+                    ${addons_uri}:mdds \
+                    ${addons_uri}:openssl \
+                    ${addons_uri}:xmlsec \
+                    ${addons_uri}:python \
+                    ${osuosl_uri}:dicts \
+                    ${osuosl_uri}:translations
+distfiles           ${main_distfile}:main \
+                    ${name}-dictionaries-7.0.3.1.tar.xz:dicts \
+                    ${name}-translations-7.0.3.1.tar.xz:translations \
+                    0168229624cfac409e766913506961a8-ucpp-1.3.2.tar.gz:ucpp \
+                    48d647fbd8ef8889e5a7f422c1bfda94-clucene-core-2.3.3.4.tar.gz:clucene-core \
+                    884ed41809687c3e168fc7c19b16585149ff058eca79acbf3ee784f6630704cc-opens___.ttf:opens \
+                    Python-3.7.7.tar.xz:python \
+                    a8c2c5b8f09e7ede322d5c602ff6a4b6-mythes-1.2.4.tar.gz:mythes \
+                    dtoa-20180411.tgz:dtoa \
+                    libabw-0.1.3.tar.xz:abw \
+                    libcmis-0.5.2.tar.xz:cmis \
+                    libe-book-0.1.3.tar.xz:ebook \
+                    libepubgen-0.1.1.tar.xz:epubgen \
+                    libexttextcat-3.4.5.tar.xz:exttextcat \
+                    libmwaw-0.3.16.tar.xz:mwaw \
+                    liborcus-0.15.4.tar.bz2:orcus \
+                    libqxp-0.0.2.tar.xz:qxp \
+                    libstaroffice-0.0.7.tar.xz:libstaroffice \
+                    libzmf-0.0.2.tar.xz:zmf \
+                    mdds-1.6.0.tar.bz2:mdds \
+                    openssl-1.0.2t.tar.gz:openssl \
```

Comment:
I will need to check if this needs to be here, as anything within is already linking to system OpenSSL. I think it can work with LibreSSL from Apple too. Prior to doing this, the external fetch step was being run and it was grabbing this file even though I was fairly certain I told it to use system OpenSSL.

Reply:
Also regarding Python, the official builds use their own internal version which then goes in `Resources` in the app bundle. From configure:

```
  --enable-python=<no/auto/system/internal/fully-internal>
                          Enables or disables Python support at run-time. Also
                          specifies what Python to use at build-time.
                          'fully-internal' even forces the internal version
                          for uses of Python during the build. On macOS the
                          only choices are 'internal' (default) or
                          'fully-internal'. Otherwise the default is 'auto'.
```

So even though I've been specifying 'system', it ignores that and uses 'internal'. So a bdep needs to be in the Portfile at least for build in that case, or if 'fully-internal' is specified it can just use the packaged version for build and runtime.

Should we pursue a patch to get system Python working at runtime? If so, should we make it pick one version of Python or do variants on the Python version (only one can be picked)? LibreOffice comes with 3.7.7 so it should be 37 or higher.

Reply:
I would have expected `--enable-python=system` to use system (MacPorts) python, and that we would be able to specify which python that is; if it does not, that feels like an upstream bug.

If we can use MacPorts python, you could either add variants for a few recent python versions like 38 and 39 and if you like 37 and default to what the python portgroup defaults to (currently 38), or just unconditionally use what the python portgroup defaults to (currently 38).

Reply:
I have made a patch to force system Python for now. Upstream would probably want something like `--enable-bogus-python` like their `--enable-bogus-pkg-config` option, only because they want to avoid if `python` in `PATH` points to `/usr/bin/python` on macOS.